.eslint/rules/icon-has-viewBox.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * @fileoverview All Paste icon svgs should have a viewBox attribute
+ * @author Glorilí Alejandro
+ */
+
+const getJSXIdentifier = (node) => {
+  if (node === undefined) {
+    return;
+  }
+  if (node.type === 'JSXIdentifier') {
+    return node;
+  }
+  if (node.name !== undefined) {
+    return getJSXIdentifier(node.name);
+  }
+  if (node.openingElement !== undefined) {
+    return getJSXIdentifier(node.openingElement);
+  }
+  return undefined;
+};
+
+module.exports = {
+  meta: {
+    type: 'problem',
+  },
+
+  create(context) {
+    return {
+      // this lets us work with all the JSXElement nodes
+      JSXElement(node) {
+        // I want to get the openingElement node of each, since that's
+        // where the details about both it being an svg and the
+        // attributes live.
+        const elementDetails = node.openingElement;
+
+        // To get find the name, we need to go into the openingElement
+        // we just got, get its JSXIdentifier, and then get that node's name
+        if ((getJSXIdentifier(elementDetails) || {}).name === 'svg') {
```

Comment:
Great question! This rule is going to look at every node that's a JSXElement in the whole repo, and for various reasons (some of those nodes might not have a JSXIdentifier bc they don't have an `openingElement` or `name`) the `getJSXIdentifier` call could be returning `undefined`. We can't do `.name` on undefined since it breaks, but we can do it on this empty object so that it will be falsey in comparison to the 'svg' string, and the code can keep running.

Lemme know if that makes sense or not, happy to keep talkin' about it!